main.go

```diff
@@ -0,0 +1,543 @@
+		return err
+	}
+	if result.Valid() {
+		log.Printf("The task payload is valid.\n")
+	} else {
+		log.Printf("The task payload is invalid. See errors:\n")
+		for _, desc := range result.Errors() {
+			log.Printf("- %s\n", desc)
+		}
+		// Dealing with Invalid Task Payloads
+		// ----------------------------------
+		// If the task payload is malformed or invalid, keep in mind that the
+		// queue doesn't validate the contents of the `task.payload` property,
+		// the worker may resolve the current run by reporting an exception.
+		// When reporting an exception, using `queue.reportException` the
+		// worker should give a `reason`. If the worker is unable execute the
+		// task specific payload/code/logic, it should report exception with
+		// the reason `malformed-payload`.
+		if err := task.reportException(); err != nil {
+			log.Printf("Error occurred reporting exception for task %v:\n%v\n", task.TaskId, err)
+		}
+		return fmt.Errorf("Validation of payload failed for task %v", task.TaskId)
+	}
+	return json.Unmarshal(jsonPayload, &task.Payload)
+}
+
+func (task *TaskRun) reportException() error {
+	ter := queue.TaskExceptionRequest{Reason: json.RawMessage(`"malformed-payload"`)}
+	tsr, callSummary := Queue.ReportException(task.TaskId, strconv.FormatInt(int64(task.RunId), 10), &ter)
+	if callSummary.Error != nil {
+		log.Printf("Not able to report exception for task %v:\n%v", task.TaskId, callSummary.Error)
+		return callSummary.Error
+	}
+	task.TaskClaimResponse.Status = tsr.Status
+	log.Println(task.String())
+	return nil
+}
+
+func (task *TaskRun) reportFailed() error {
+	tsr, callSummary := Queue.ReportFailed(task.TaskId, strconv.FormatInt(int64(task.RunId), 10))
+	if callSummary.Error != nil {
+		log.Printf("Not able to report failed completion for task %v:\n%v", task.TaskId, callSummary.Error)
+		return callSummary.Error
+	}
+	task.TaskClaimResponse.Status = tsr.Status
+	log.Println(task.String())
+	return nil
+}
+
+func (task *TaskRun) reportCompleted() error {
+	log.Printf("Command finished successfully!")
+	tsr, callSummary := Queue.ReportCompleted(task.TaskId, strconv.FormatInt(int64(task.RunId), 10))
+	if callSummary.Error != nil {
+		log.Printf("Not able to report successful completion for task %v:\n%v", task.TaskId, callSummary.Error)
+		return callSummary.Error
+	}
+	task.TaskClaimResponse.Status = tsr.Status
+	log.Println(task.String())
+	return nil
+}
+
+func (task *TaskRun) run() error {
+
+	fmt.Println("Running task!")
+	fmt.Println(task.String())
+	task.prepEnvVars()
+	cmd := exec.Command(task.Payload.Command[0], task.Payload.Command[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+	log.Printf("Waiting for command to finish...")
+	// start a go routine to kill task after max run time...
+	go func() {
+		time.Sleep(time.Second * time.Duration(task.Payload.MaxRunTime))
+		task.cancel("max run time (" + strconv.Itoa(task.Payload.MaxRunTime) + "s) exceeded")
```

Comment:
Probably better named abort... but I guess this is just `cmd.Process.Kill()`.

When you add logging support it might be nice if you after killing writing a message to the log, saying that it was killed...